core/logger/splunk.go

```diff
@@ -0,0 +1,247 @@
+package logger
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/core/logger/audit"
+
+	"go.uber.org/zap/zapcore"
+)
+
+type splunkLogger struct {
+	logger          Logger
+	splunkToken     string
+	splunkURL       string
+	environmentName string
+	hostname        string
+	localIP         string
+}
+
+func newSplunkLogger(logger Logger, splunkToken string, splunkURL string, hostname string, environment string) Logger {
+	// Initialize and return Splunk logger struct with required state for HEC calls
+	return &splunkLogger{
+		logger:          logger.Helper(1),
+		splunkToken:     splunkToken,
+		splunkURL:       splunkURL,
+		environmentName: environment,
+		hostname:        hostname,
+		localIP:         getLocalIP(),
+	}
+}
+
+func (l *splunkLogger) Audit(eventID audit.EventID, data map[string]interface{}) {
+	// goroutine to async POST to splunk HTTP Event Collector (HEC)
+	go l.postLogToSplunk(eventID, data)
```

Comment:
Yeah, I don't like this. We have unbounded goroutines here. Would prefer to see a buffer and loop with start/stop functions.